### PR TITLE
Bug/headers

### PR DESCRIPTION
**Pull request by jrconlin**
- Address bug 1100004 and related issues.

@kitcambridge r?

### CODE REVIEW

File: wmf/handlers.go [hunk 1]
```diff
@@ -2047,9 +2053,25 @@ func (r *Handler) Language(resp http.ResponseWriter, req *http.Request) {
...
 }
 
+func (self *Handler) SecureHeaders(resp http.ResponseWriter) {
+	if resp != nil {
+		resp.Header().Set("Strict-Transport-Security", "max-age=15552000")
+		resp.Header().Set("X-Frame-Options", "deny")
+		resp.Header().Set("Frame-Options", "deny")
+		resp.Header().Set("X-XSS-Protection", "1; mode=block")
+		resp.Header().Set("X-Content-Type-Options", "nosniff")
+		resp.Header().Set("X-Content-Security-Policy", "Content-Security-Policy: default-src 'self'")
```
Comment (kitcambridge) [inline wmf/handlers.go]: Duplicate nested header (`X-Content-Security-Policy: Content-Security-Policy: default-src 'self'`), or intentional?

File: wmf/handlers.go [hunk 2]
```diff
@@ -2259,17 +2282,20 @@ func (self *Handler) Static(resp http.ResponseWriter, req *http.Request) {
 	/* This should be handled by something like an nginx rule
 	 */
 
+	self.SecureHeaders(resp)
 	if !self.config.GetFlag("use_insecure_static") {
 		return
 	}
-
-	http.ServeFile(resp, req, self.docRoot+req.URL.Path)
+	filePath := path.Clean(self.docRoot + req.URL.Path)
```
Comment (kitcambridge) [inline wmf/handlers.go]: Nice catch!

**Comment by kitcambridge** [comment]
Looks great; merge away!

File: wmf/handlers.go (hunk 1 above)
Comment (jrconlin) [inline wmf/handlers.go]: Nope, misread the example form from https://www.owasp.org/index.php/List_of_useful_HTTP_headers

File: wmf/handlers.go (hunk 2 above)
Comment (jrconlin) [inline wmf/handlers.go]: thus why the option was "use_insecure_static", but it's a bit less insecure now.

**Comment by SymbianSyMoh** [comment]
Great!

**Pull request merged**